modules/account-lib/src/coin/sol/keyPair.ts

```diff
@@ -67,4 +68,19 @@ export class KeyPair extends Ed25519KeyPair {
     const keys = this.getKeys();
     return keys.pub;
   }
+
+  /**
+   * Derives a hardened child key pair using this key pair's secret key
+   * as the seed.
+   *
+   * @param path derivation path
+   */
+  deriveHardened(path: string): KeyPair {
```

Comment:
If this is using the raw seed then we can put it in the ed25519 base class, no?

Reply:
good point, you're right. i moved to ed25519 key pair and return DefaultKeys instead of the actual key pair.